docs(self-hosted configuration): update binarySource (#22619)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

## binarySource

Renovate often needs to use third-party binaries in its PRs, like `npm` to update `package-lock.json` or `go` to update `go.sum`.

> By default, Renovate uses a child process to run such binaries, so they must be:

Author: We need to decide if we want to use `binaries` or `tools` in the docs. Right now we use `tools` and `binaries`, which is a bit confusing. 😄

- installed before running Renovate
- available in the path

But you can tell Renovate to use "sidecar" containers for third-party binaries by setting `binarySource=docker`.
For this to work, `docker` needs to be installed and the Docker socket available to Renovate.
Now Renovate uses `docker run` to create containers like Node.js or Python to run binaries in as-needed.

Additionally, when Renovate is run inside a container built using [`containerbase`](https://github.com/containerbase), such as the official Renovate images on Docker Hub, then `binarySource=install` can be used.

> This mode means that Renovate will dynamically install the version of binaries available, if supported. // I think we now support install-mode for all binaries/tools???

Author: This line needs a suggestion from the maintainers. 😉

> If all projects are managed by Hermit, you can tell Renovate to use the binary versions specified in each project via Hermit by setting `binarySource=hermit`.

Author: The Hermit sentence feels really clunky.

> Binaries not on this list fall back to `binarySource=global`.

Author: We don't have a list any more, so what should the sentence be now?

Now Renovate uses `docker run` to create containers like Node.js or Python to run tools in as-needed.

Additionally, when Renovate is run inside a container built using [`containerbase`](https://github.com/containerbase), such as the official Renovate images on Docker Hub, then `binarySource=install` can be used.

> This mode means that Renovate will dynamically install the version of tools available, if supported. // I think we now support install-mode for all tools???

Author: Can you suggest a change to this line, I don't know how to change it now we support `install-mode` for all tools? 😕

> This mode means that Renovate will dynamically install the version of tools available, if supported. // I think we now support install-mode for all tools???

Reviewer: ```suggestion
This mode means that Renovate will dynamically install the desired version of each tool needed.
```

> If all projects are managed by Hermit, you can tell Renovate to use the tool versions specified in each project via Hermit by setting `binarySource=hermit`.

Author: The Hermit sentence feels clunky, and a bit out of place.

Also I would expect to see an example of the correct Hermit configuration, maybe?

Reviewer: I'm ok to leave this as-is

> Tools not on this list fall back to `binarySource=global`.

Author: We don't have a list any more, so what should the sentence be now?

> Tools not on this list fall back to `binarySource=global`.

Reviewer: ```suggestion
```
